gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -507,42 +508,93 @@ public static Builder builder() {
       private final Set<BlobTargetOption> targetOptions = new LinkedHashSet<>();
       private BlobId source;
       private BlobInfo target;
+      private Long megabytesRewrittenPerCall;
 
+      /**
+       * Sets the blob to rewrite given bucket and blob name.
+       *
+       * @return the builder.
+       */
       public Builder source(String bucket, String blob) {
         this.source = BlobId.of(bucket, blob);
         return this;
       }
 
+      /**
+       * Sets the blob to rewrite given a {@link BlobId}.
+       *
+       * @return the builder.
+       */
       public Builder source(BlobId source) {
         this.source = source;
         return this;
       }
 
+      /**
+       * Sets blob's source options.
+       *
+       * @return the builder.
+       */
       public Builder sourceOptions(BlobSourceOption... options) {
         Collections.addAll(sourceOptions, options);
         return this;
       }
 
+      /**
+       * Sets blob's source options.
+       *
+       * @return the builder.
+       */
       public Builder sourceOptions(Iterable<BlobSourceOption> options) {
         Iterables.addAll(sourceOptions, options);
         return this;
       }
 
+      /**
+       * Sets the rewrite target.
+       *
+       * @return the builder.
+       */
       public Builder target(BlobInfo target) {
         this.target = target;
         return this;
       }
 
+      /**
+       * Sets blob's target options.
+       *
+       * @return the builder.
+       */
       public Builder targetOptions(BlobTargetOption... options) {
         Collections.addAll(targetOptions, options);
         return this;
       }
 
+      /**
+       * Sets blob's target options.
+       *
+       * @return the builder.
+       */
       public Builder targetOptions(Iterable<BlobTargetOption> options) {
         Iterables.addAll(targetOptions, options);
         return this;
       }
 
+      /**
+       * Sets the maximum number of megabytes to copy for each RPC call. This parameter is ignored
+       * if source and target blob share the same location and storage class as rewrite is made with
```

Comment:
We should probably avoid using the "rewrite" term in the user-api (and use "copy") instead.